csrc/scheduler/reduction_heuristic.h

```diff
@@ -309,7 +318,8 @@ class ReductionParams : public HeuristicParams {
   }
 
   std::shared_ptr<HeuristicParams> clone() const override {
-    return std::make_shared<ReductionParams>(*this);
+    auto ptr = std::make_shared<ReductionParams>(*this);
+    return std::static_pointer_cast<HeuristicParams>(ptr);
```

Comment:
Why is this change?

Reply:
clang tidy requires this explicit type cast.